src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,111 @@
+    let relativeActivityContext = useContext(RelativeActivityContext);
+
+    let dataTypes: RelativeActivityDataType[] = props.dataTypes || [];
+    if (props.useContext && relativeActivityContext?.dataTypes) {
+        dataTypes = relativeActivityContext.dataTypes;
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
+
+    function loadData() {
+        function transformResults(results: { [key: string]: { [key: string]: RelativeActivityQueryResult } } | undefined) {
+            if (!results) return;
+            let computedResult: { [key: string]: RelativeActivityQueryResult } = {};
+            dataTypes.forEach(dataType => {
+                computedResult[dataType.dailyDataType] = results[dataType.dailyDataType][getDayKey(date!)];
+            });
+            return computedResult;
+        }
+
+        setResults(undefined);
+        if (props.useContext) {
+            setResults(transformResults(relativeActivityContext!.data));
+            return;
+        }
+        queryRelativeActivity(date!, date!, dataTypes, !!props.previewState).then(results => {
+            setResults(transformResults(results));
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, relativeActivityContext, props.date, props.dataTypes], 0);
+
+    if (results == null || !Object.keys(results).length) {
+        return null;
+    }
+
+    function getBarColor(dataType: RelativeActivityDataType, fillPercent: number) {
+        if (dataType.threshold !== undefined && dataType.overThresholdColor && fillPercent > .5) {
+            return dataType.overThresholdColor;
+        }
+        return dataType.color || "var(--mdhui-color-primary)";
+    }
+
+    function getThresholdLabel(dataType: RelativeActivityDataType, threshold: number) {
+        if (!dataTypes.some(d => d.threshold !== undefined)) {
+            return undefined;
+        }
+
+        let dataTypeDefinition = getDailyDataTypeDefinition(dataType.dailyDataType);
+        return dataTypeDefinition.formatter(threshold);
+    }
+
+    return <div ref={props.innerRef} className="mdhui-relative-activity">
+        {props.title && <CardTitle title={props.title} />}
+        {dataTypes.map(d => {
+            let dataTypeResult = results![d.dailyDataType];
+            if (!dataTypeResult) {
+                return null;
+            }
+            let dataTypeDefinition = getDailyDataTypeDefinition(d.dailyDataType);
+
+            return <div key={d.dailyDataType} className="mdhui-relative-activity-datatype">
+                <ActivityMeter className="mdhui-relative-activity-meter"
+                    label={dataTypeDefinition.getLabel()}
+                    value={dataTypeDefinition.formatter(dataTypeResult.value) || dataTypeResult.value.toString()}
```

Comment:
Do we really want this defaulting?  It may be unusual, but if the formatter returns something like empty string for a given value, maybe we should respect that rather than displaying the non-formatted value?

Reply:
Ah yep, good call, this is from before when I was making formatters optional (which would have been wrong then too)

Reply:
Oh wait - nm, formatters are still optional if you call the legacy registerDailyDataProvider - so yeah I just need to fix this to default to value if there is no formatter

Reply:
Fixed